amethyst_network/src/network_socket.rs

```diff
@@ -66,27 +66,31 @@ where
         let udp_send_handle = server.udp_send_handle();
         let udp_receive_handle = server.udp_receive_handle();
 
-        let server_sender = NetSocketSystem::<E>::start_sending(udp_send_handle);
-        let server_receiver = NetSocketSystem::<E>::start_receiving(udp_receive_handle);
+        let event_sender = NetSocketSystem::<E>::start_sending(udp_send_handle);
 
         Ok(NetSocketSystem {
             filters,
-            transport_sender: server_sender,
-            transport_receiver: server_receiver,
+            event_sender,
+            event_receiver: udp_receive_handle,
             config,
         })
     }
 
     /// Start a thread to send all queued packets.
     fn start_sending(sender: Sender<Packet>) -> Sender<InternalSocketEvent<E>> {
-        let (tx, send_queue) = crossbeam_channel::unbounded();
+        let (event_sender, event_receiver) = crossbeam_channel::unbounded();
 
         thread::spawn(move || loop {
-            for control_event in send_queue.try_iter() {
+            for control_event in event_receiver.try_iter() {
                 match control_event {
                     InternalSocketEvent::SendEvents { target, events } => {
                         for ev in events {
-                            send_event(ev, target, &sender);
+                            match ev {
+                                NetEvent::Packet(packet) => {
+                                    send_event(packet, target, &sender);
+                                }
+                                _ => { /* TODO, handle connect, disconnect etc. */ }
```

Comment:
If we aren't handling connect/disconnect, doesn't this mean amethyst_network won't work at all?

Reply:
exactly

Reply:
Shouldn't we handle that?

Reply:
Yes, we should be, but that's the task for an other PR.